### DISCUSSION TITLE
css question

### DISCUSSION THREAD

**Discussion by N3C2L** [madprops/grasshopper#65] [General]
I've changed my taglist a little with css to this:

<img width="240" alt="taglist" src="https://github.com/madprops/grasshopper/assets/74585843/bbc57f75-2805-4e6e-9379-4d6e164ba946">

But now I want to move the favicon to be in the same line as the tab title above.
I found this css selector .item_icon, but it selects every favicon not just the one of the favicon in the taglist.

Is there another css selector I miss to move this favicon more higher?

**Comment by Kecbm** [comment]
Hello @N3C2L, for align the five icon and the text in the same line you can use the display flex. For it, you will envolve the content whit a div, with the id="content" for example, and aply the css:

#content {
display: flex;
align-items: center;
}

This modification should centralize the content. After applicate it, coment here if it worked 😁

**Comment by N3C2L** [comment]
Thank you, but I think you misread my post. I wasn't asking about aligning five icons , instead I asked for moving the favicon of the website to be the same height with the tab title.

**Comment by madprops** [comment]
I'll reply to this soon. You'll need a new class for tagged items.

**Comment by N3C2L** [comment]
Thanks!

**Comment by madprops** [comment]
So you can do this now:

```css
.using_taglist_below > div:not(.item_content) {
  margin-top: -20px;
}
```

This will apply that to all divs except the text. You might have to adjust it.
Aligning to flex-start would move all items to the very top, which is not what you want.
Though you could `align-items: flex-start` and add some margin adjustments from there.

**Reply by N3C2L** [reply]
Ok thanks, I will try it out.

**Reply by N3C2L** [reply]
I tried it with your code above, but the hover button will be too big then and overlaps the tab above. Instead, it would be very nice, if you could implement a setting for this alignment in the taglist area.
I found another bug regarding the hover button: how to reproduce 1. select a tab that has no color and no tag 2. Try to apply a tag/color cmd with the hover button 3. You need to run the command a second time (executed via the hover button) to get the tag/color visible on the tab.

**Reply by madprops** [reply]
Ok the bug gets fixed in next version. Not sure i'll add that setting though, unless it's simple enough.

**Reply by N3C2L** [reply]
Ok thanks, you are right I maybe I don't need the setting if you could tell me the class to reduce the size of the hover button.

**Reply by madprops** [reply]
Maybe something like:

```css
.using_taglist_below > div:not(.item_content):not(.hover_button):not(.close_button) {
  margin-top: -20px;
}
```

**Reply by madprops** [reply]
Or:

```css
.using_taglist_below > div:not(.item_content):not(.hover_button):not(.close_button) {
  align-items: flex-start;
  margin-top: 20px;
}
```

**Reply by N3C2L** [reply]
thanks, I will try them.

**Reply by N3C2L** [reply]
The second one did it, thanks a lot.

**Comment by dharmi04** [comment]
<span> tag can also work.

**Comment by N3C2L** [comment]
@madprops 
May I ask you if there is a css class for the grasshopper popup? Because I want to style it different than the GH sidebar. Background: I want to put the footer on top in the GH popup. Why? Because then I can spot the URLs faster when I am in the History or Bookmarks Mode by hovering over the items.

**Reply by madprops** [reply]
How about I give you a setting to put footer on top. CSS would be a bit messy. Let me see if I can do it.

Also there's no css selectors for sidebar or popup, they are treated the same.

![footerontop](https://github.com/user-attachments/assets/36bd64c8-658f-490c-aaec-842a94d30685)

Upload soon. Need to fix some css margins.

And without title:

![footertop](https://github.com/user-attachments/assets/b6184e81-51d8-4a3d-8a96-641b641d0eac)

**Reply by N3C2L** [reply]
Hmm, thanks at first. But the problem is, in the sidebar I would want to have the footer completly hidden because I need the space. Do you think an additional setting would be possible to have the footer on top only in the GH popup, but not in the sidebar?

**Reply by madprops** [reply]
I'll see if that's possible

**Reply by N3C2L** [reply]
Thanks!

**Reply by madprops** [reply]
Well there's the possibility of using the css selectors that apply depending on height:

```css
@media screen and (max-height:800px)
{
  .footer {
    height: 6.5rem !important;
  }
}
```

That one would apply to the popup only implying it's less than 800px in height.

You can change that to min-height to target sidebar etc.

**Reply by N3C2L** [reply]
Thanks, I will give it a try.
